src/buildlog/mod.rs

```diff
@@ -51,10 +52,46 @@ impl BuildLog {
 		}
 	}
 
+	pub fn add_entry(&mut self, output: RawString, build_starttime: Instant, starttime: Instant, endtime: Instant, command: RawString) {
+		self.entries.insert(output, Entry {
+			start_time_ms: (starttime - build_starttime).as_millis() as u32,
```

Comment:
`Instant - Instant` panics if the second one is larger. Can that ever happen in some edge case? If so, might want to just put `0` in `start_time_ms`/`end_time_ms` in that case.